packages/wrangler/src/triggers/deploy.ts

```diff
@@ -94,55 +83,17 @@ export default async function triggersDeploy(
 	const uploadMs = Date.now() - start;
 	const deployments: Promise<string[]>[] = [];
 
-	const deploymentInSync = deployToWorkersDev === available_on_subdomain;
-	const previewsInSync =
-		config.preview_urls === previews_available_on_subdomain;
-
-	if (deployToWorkersDev) {
-		// Deploy to a subdomain of `workers.dev`
-		const userSubdomain = await getWorkersDevSubdomain(
-			config,
-			accountId,
-			config.configPath
-		);
-
-		const deploymentURL =
-			props.legacyEnv || !props.env
-				? `${scriptName}.${userSubdomain}`
-				: `${envName}.${scriptName}.${userSubdomain}`;
-
-		if (deploymentInSync && previewsInSync) {
-			deployments.push(Promise.resolve([deploymentURL]));
-		} else {
-			// Enable the `workers.dev` subdomain.
-			deployments.push(
-				fetchResult(config, `${workerUrl}/subdomain`, {
-					method: "POST",
-					body: JSON.stringify({
-						enabled: true,
-						previews_enabled: config.preview_urls,
-					}),
-					headers: {
-						"Content-Type": "application/json",
-					},
-				}).then(() => [deploymentURL])
-			);
-		}
-	}
-	if (!deployToWorkersDev && (!deploymentInSync || !previewsInSync)) {
-		// Disable the workers.dev deployment
-		await fetchResult(config, `${workerUrl}/subdomain`, {
-			method: "POST",
-			body: JSON.stringify({
-				enabled: false,
-				previews_enabled: config.preview_urls,
-			}),
-			headers: {
-				"Content-Type": "application/json",
-			},
-		});
-	}
-	if (!deployToWorkersDev && deploymentInSync && routes.length !== 0) {
+	const { wantWorkersDev, workersDevInSync } = await subdomainDeploy(
+		props,
+		accountId,
+		scriptName,
+		envName,
+		workerUrl,
+		routes,
+		deployments
+	);
+
+	if (!wantWorkersDev && workersDevInSync && routes.length !== 0) {
```

Comment:
There was a bug in our testing suite that was always setting `workers_dev = true`, causing this block to never execute:
https://github.com/cloudflare/workers-sdk/pull/10478/files#r2309034991
Now that it does execute, tests mocks were updated accordingly.